packages/synapse-react-client/src/components/ChallengeRegisterButton/ChallengeRegisterButton.tsx

```diff
@@ -135,17 +100,19 @@ const ChallengeRegisterButton = ({
         <SpinnerButton
           disableElevation={true}
           variant="contained"
-          color="secondary"
-          onClick={() => (onJoinClick ? onJoinClick() : undefined)}
+          onClick={() =>
+            accessToken && onJoinClick ? onJoinClick() : undefined
```

Comment:
The challenge register button already handles the case when the user is not logged in (it pops up a sign-in dialog), so don't call back until user is logged in.  In this case, after login, it will call back and cause the AccessRequirementList dialog to pop up (which will at least say that you need to be logged in!)